### ISSUE TITLE
Error model predict method

### ISSUE THREAD

**Issue by purvang3** [Deci-AI/super-gradients#1166]
### 🐛 Describe the bug

After successful training, I am trying to do prediction on locally stored image but getting following error.

```
[2023-06-13 04:18:01] INFO - checkpoint_utils.py - Successfully loaded model weights from my_first_yolonas_run/ckpt_best.pth EMA checkpoint.
[2023-06-13 04:18:01] INFO - pipelines.py - Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`
Traceback (most recent call last):
  File "/root/models/yn/train.py", line 260, in <module>
    print(best_model.predict(img_in, conf=0.25).show())
  File "/usr/local/lib/python3.9/site-packages/super_gradients/training/models/detection_models/customizable_detector.py", line 177, in predict
    return pipeline(images)  # type: ignore
  File "/usr/local/lib/python3.9/site-packages/super_gradients/training/pipelines/pipelines.py", line 96, in __call__
    return self.predict_images(inputs, batch_size)
  File "/usr/local/lib/python3.9/site-packages/super_gradients/training/pipelines/pipelines.py", line 111, in predict_images
    return self._combine_image_prediction_to_images(result_generator, n_images=len(images))
  File "/usr/local/lib/python3.9/site-packages/super_gradients/training/pipelines/pipelines.py", line 290, in _combine_image_prediction_to_images
    images_predictions = [next(iter(images_predictions))]
  File "/usr/local/lib/python3.9/site-packages/super_gradients/training/pipelines/pipelines.py", line 149, in _generate_prediction_result
    yield from self._generate_prediction_result_single_batch(batch_images)
  File "/usr/local/lib/python3.9/site-packages/super_gradients/training/pipelines/pipelines.py", line 176, in _generate_prediction_result_single_batch
    model_output = self.model(torch_inputs)
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/super_gradients/training/models/detection_models/customizable_detector.py", line 87, in forward
    x = self.backbone(x)
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/super_gradients/modules/detection_modules.py", line 80, in forward
    x = getattr(self, layer)(x)
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/super_gradients/training/models/detection_models/yolo_nas/yolo_stages.py", line 138, in forward
    return self.conv(x)
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/super_gradients/modules/qarepvgg_block.py", line 179, in forward
    return self.se(self.nonlinearity(self.rbr_reparam(inputs)))
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/conv.py", line 463, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/usr/local/lib/python3.9/site-packages/torch/nn/modules/conv.py", line 459, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: Input type (unsigned char) and bias type (float) should be the same
```
**code snippet**
```
    best_model = models.get('yolo_nas_l',
                        num_classes=len(dataset_params['classes']),
                        checkpoint_path="my_first_yolonas_run/ckpt_best.pth")
                        
    img_in = "www.mynumi.net-USASE5AD160-31.jpg"
    print(best_model.predict(img_in, conf=0.25).show())
```

following [inference](https://colab.research.google.com/drive/1q0RmeVRzLwRXW-h9dPFSOchwJkThUy6d#scrollTo=iqXzphK6XXTF&line=2&uniqifier=1) working fine. 

not sure why error occurs only with trained model on custom dataset.

### Versions

```
Core(s) per socket:              8
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           167
Model name:                      11th Gen Intel(R) Core(TM) i9-11900K @ 3.50GHz
Stepping:                        1
CPU MHz:                         3474.617
CPU max MHz:                     5300.0000
CPU min MHz:                     800.0000
BogoMIPS:                        7008.00
Virtualization:                  VT-x
L1d cache:                       384 KiB
L1i cache:                       256 KiB
L2 cache:                        4 MiB
L3 cache:                        16 MiB
NUMA node0 CPU(s):               0-15
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx avx512f avx512dq rdseed adx smap avx512ifma clflushopt intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq rdpid md_clear flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] numpy==1.23.0
[pip3] torch==2.0.1
[pip3] torchmetrics==0.8.0
[pip3] torchvision==0.15.2
[pip3] triton==2.0.0
[conda] Could not collect
```

**Comment by whittenator** [comment]
@purvang3 I am also facing this issue. Any luck fixing it?

**Comment by whittenator** [comment]
@purvang3 I have found the solution to the issue, it seems that there has been a bug introduced in the latest versions of super-gradients. Downgrading like so solved my issue: `pip install super-gradients==3.1.1`

**Comment by ofrimasad** [comment]
Hi @purvang3 .
are you trying to predict on an image from the same data you have used for training?

**Comment by whittenator** [comment]
> Hi @purvang3 .
> 
> are you trying to predict on an image from the same data you have used for training? 

@ofrimasad I was using the test images from my dataset when I received the same error as @purvang3

**Comment by Gabriellgpc** [comment]
same issue here.

**Comment by shooorf** [comment]
**Stupid Workaround:**

File  /usr/local/lib/python3.8/dist-packages/super_gradients/training/pipelines/pipelines.py
(change path to your environment specific)
Function `_generate_prediction_result_single_batch`
Line 175: `preprocessed_images.append(preprocessed_image)`
Change to `preprocessed_images.append(preprocessed_image.astype(np.single))`

If you used `coco_detection_yolo_format_train` and `coco_detection_yolo_format_val `functions from example notebook:
you model was trained on 0-255 float input (not 0-1) because of incorrect default preprocessng, so i guess you obtained sub-optimal results.

**Comment by Gabriellgpc** [comment]
@shooorf  how can I see that list of preprocessings from my checkpoint ?

**Comment by shooorf** [comment]
> @shooorf how can I see that list of preprocessings from my checkpoint ?

from super_gradients.training.utils.checkpoint_utils import read_ckpt_state_dict
stdict = read_ckpt_state_dict(ckpt_path=checkpoint_path)
print(stdict['processing_params']['image_processor'])

**Comment by shivarajkarki** [comment]
Hi Folks,
I was getting similar kind of error, try fixing it at `src/supergradients/training/pipelines/pipelines.py` inside funtion "_generate_prediction_result_single_batch" try checking dtype of torch_input at line 173 then you can change or typecast it.

**Comment by BloodAxe** [comment]
Fixed in 3.1.3
See
* https://github.com/Deci-AI/super-gradients/releases/tag/3.1.3
* https://github.com/Deci-AI/super-gradients/pull/1281
